substrate/frame/support/procedural/src/pallet/expand/hooks.rs

```diff
@@ -140,6 +140,66 @@ pub fn expand_hooks(def: &mut Def) -> proc_macro2::TokenStream {
 		}
 	};
 
+	let pre_upgrade_post_upgrade = if cfg!(feature = "try-runtime") {
+		quote::quote_spanned!(span =>
+			fn pre_upgrade() -> Result<#frame_support::__private::Vec<u8>, #frame_support::sp_runtime::TryRuntimeError> {
+				<
+					Self
+					as
+					#frame_support::traits::Hooks<#frame_system::pallet_prelude::BlockNumberFor::<T>>
+				>::pre_upgrade()
+			}
+
+			fn post_upgrade(state: #frame_support::__private::Vec<u8>) -> Result<(), #frame_support::sp_runtime::TryRuntimeError> {
+				#post_storage_version_check
+
+				<
+					Self
+					as
+					#frame_support::traits::Hooks<#frame_system::pallet_prelude::BlockNumberFor::<T>>
+				>::post_upgrade(state)
+			}
+		)
+	} else {
+		proc_macro2::TokenStream::new()
+	};
+
+	let try_state = if cfg!(feature = "try-runtime") {
```

Comment:
We should not do this. This is partly "cursed". Sometimes this is not really working well.

You can create a macro in `frame_support` using [this macro](https://github.com/paritytech/polkadot-sdk/blob/9bd1f252625a6de1a58c6cca6b7e9573b2b3782f/substrate/primitives/core/src/lib.rs#L433) and then you can use this to guard the code. [Here](https://github.com/paritytech/polkadot-sdk/commit/d1b0ef76a8b060437ec7dfc1bf6b400626cd6208) is an example of what I mean.

Reply:
ah yes, I implemented as you suggested here https://github.com/paritytech/polkadot-sdk/pull/5099/commits/61fa26dff59b65d1bf1fc98c5635e0387595804a